### ISSUE TITLE
`Microsoft Entra ID authentication with Kubernetes RBAC` not working though aadprofile is enabled with adminGroupObjectIDs

### ISSUE THREAD

**Issue by RSE132** [kubernetes-sigs/cluster-api-provider-azure#4891]
### What steps did you take and what happened?

I have been trying to a deploy a managed aks cluster. The deployment happened properly but the aad integration is not working properly

I have following parameters enabled under `AzureManagedControlPlane`

```
  aadProfile:
    managed: true
    adminGroupObjectIDs: 
    - fe7b0187-xxxxx-xxxxxx-xxxxxx-2f58fxxxxxx
  disableLocalAccounts: false
```

But when I look through azure portal I see it is enabling `Microsoft Entra ID authentication with azure RBAC` instead of `Microsoft Entra ID authentication with Kubernetes RBAC`

### What did you expect to happen?

I expected on the azure portal it will show me `Microsoft Entra ID authentication with Kubernetes RBAC` for the cluster instead I see this -

![image](https://github.com/kubernetes-sigs/cluster-api/assets/70406583/e2684717-865d-460d-88f3-d7660ff3d4cc)

### Cluster API version

cluster.x-k8s.io/v1beta1

### Kubernetes version

v1.28.9

### Label(s) to be applied

/kind bug
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

**Comment by RSE132** [comment]
any update on this please ?

**Comment by RSE132** [comment]
`AzureManagedControlPlane` generates `ManagedCluster` resource which has a field `enableAzureRBAC` and this default set to `true`. Is there a way that we can define `enableAzureRBAC=false` inside `AzureManagedControlPlane` ?

**Comment by willie-yao** [comment]
Hi @RSE132! I took a look and it looks like `enableAzureRBAC` is set to the `managed` field in the `AADProfile`: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/56259aa79f3702586c0d8342233ef36213a1910d/azure/scope/managedcontrolplane.go#L594

I'm not sure why `enableAzureRBAC` is set to false in your case. I'll try and reproduce this issue and get back to you. Apologies for the delay!

I also don't see a problem with allowing `enableAzureRBAC` to be configurable.

**Comment by RSE132** [comment]
@willie-yao  & @nojnhuh I think I figured it out. In order to have Microsoft Entra ID authentication with Kubernetes RBAC, `enableAzureRBAC` need to be set to `false`   
```
      aadProfile:    
        managed: true                                         
        adminGroupObjectIDs:                                  # AAD security Group
          - xxxxxx-xxxxx-xxxxx-xxxxx-xxxxxx
        enableAzureRBAC: false

**Comment by willie-yao** [comment]
Awesome! I have closed #4914 now that this is working and the PR may cause some backwards-compatibility issues.
